programs/cleos/main.cpp

```diff
@@ -482,6 +482,21 @@ void print_action( const fc::variant& at ) {
    */
    if( args.size() > 100 ) args = args.substr(0,100) + "...";
    cout << "#" << std::setw(14) << right << receiver << " <= " << std::setw(28) << std::left << (code +"::" + func) << " " << args << "\n";
+
+   std::string return_value, return_value_prefix{"return value: "};
+   const auto  & iter = at.get_object().find("return_value_data");
+   if( iter != at.get_object().end() ) {
+      return_value = fc::json::to_string(iter->value(), fc::time_point::maximum());
+   } else {
```

Comment:
Wondering if you have to check "return_value_hex_data", too?

Reply:
No, `"return_value_hex_data"` will always be present(#9354)

Reply:
What if you interact with an older version of nodeos? You need to support older nodeos versions.

Reply:
added a check for return_value_hex_data as well